ql/src/java/org/apache/hadoop/hive/ql/io/LlapCacheOnlyInputFormatInterface.java

```diff
@@ -25,4 +25,11 @@
 /** Marker interface for LLAP IO. */
 public interface LlapCacheOnlyInputFormatInterface {
   void injectCaches(FileMetadataCache metadataCache, DataCache dataCache, Configuration cacheConf);
+
+  /**
+   * For inputformats that can only accept LLAP caching with vectorization turned on.
+   */
+  interface VectorizedOnly extends LlapCacheOnlyInputFormatInterface {
```

Comment:
Why do we need vectorization turned on for LLAP caching

Reply:
Although LLAP can support reading while vectorization is turned off for simple ORC files, we don't intend to support the same for Iceberg backed ORC files. It would be a too great additional complexity for little benefit.